src/PollingBlockTracker.test.ts

```diff
@@ -0,0 +1,3024 @@
+            {
+              provider: {
+                stubs: [
+                  {
+                    methodName: 'eth_blockNumber',
+                    response: {
+                      result: '0x0',
+                    },
+                  },
+                  {
+                    methodName: 'eth_blockNumber',
+                    response: {
+                      result: '0x1',
+                    },
+                  },
+                ],
+              },
+            },
+            async ({ blockTracker }) => {
+              const receivedBlockNumbers: string[] = [];
+
+              await new Promise<void>((resolve) => {
+                setTimeoutRecorder.onNumAutomaticCallsExhausted(resolve);
+
+                blockTracker[methodToAddListener](
+                  'latest',
+                  (blockNumber: string) => {
+                    receivedBlockNumbers.push(blockNumber);
+                  },
+                );
+              });
+
+              expect(receivedBlockNumbers).toStrictEqual(['0x0', '0x1']);
+            },
+          );
+        });
+
+        it('should change the Timeout object created when the poll loop waits for the next iteration such that Node does not have to wait for it to finish before exiting', async () => {
+          const setTimeoutRecorder = recordCallsToSetTimeout();
+          const blockTrackerOptions = {
+            pollingInterval: 100,
+            blockResetDuration: 200,
+            keepEventLoopActive: false,
+          };
+
+          await withPollingBlockTracker(
+            {
+              provider: {
+                stubs: [
+                  {
+                    methodName: 'eth_blockNumber',
+                    response: {
+                      result: '0x0',
+                    },
+                  },
+                  {
+                    methodName: 'eth_blockNumber',
+                    response: {
+                      result: '0x1',
+                    },
+                  },
+                ],
+              },
+              blockTracker: blockTrackerOptions,
+            },
+            async ({ blockTracker }) => {
+              blockTracker[methodToAddListener]('latest', EMPTY_FUNCTION);
+
+              await new Promise((resolve) => {
+                blockTracker.on('_waitingForNextIteration', resolve);
+              });
+
+              const nextIterationTimeout = setTimeoutRecorder.calls.find(
+                (call) => {
+                  return call.duration === blockTrackerOptions.pollingInterval;
+                },
+              );
+              expect(nextIterationTimeout).not.toBeUndefined();
+              expect(nextIterationTimeout?.timeout.hasRef()).toBe(false);
+            },
+          );
+        });
+
+        it('should take a listener that is not called after the latest block is fetched if the new block number is less than the current block number', async () => {
```

Comment:
This one was difficult to understand as well. I think this is equivalent
```suggestion
        it('should not emit latest event if new block number is less than the current block number', async () => {
```

Reply:
Another example here of the same wording: https://github.com/MetaMask/eth-block-tracker/pull/103/files#diff-59b2763dcfc8532df91ccd2df5bfd3051d465e6eef759260d2430f4921d28c75R1425

Reply:
There are a ton of "should take a listener" tests because `on('latest', ...)` is the subject of the test here and not the block tracker itself. Would it make sense to use passive tense? Maybe it can be understood that if we are listening to "latest", "latest" can also be the subject? So, something like:

``` suggestion 
        it('should not be emitted if the new block number is less than the current block number', async () => {
```

Reply:
Hmm. Yeah that makes sense! This wording seems fine to me as well.

Reply:
Ended up going with your wording. Changed in [a3bf8ed](https://github.com/MetaMask/eth-block-tracker/pull/103/commits/a3bf8edb367ca7dff27ea5f83ece8cd75bf9d554).